surveys/views.py

```diff
@@ -53,6 +53,23 @@ def form_valid(self, form):
         return HttpResponseRedirect(self.get_success_url())
 
 
+class SurveyPublish(TemplateView):
+    template_name = "survey_publish.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['survey'] = SurveyFormEntry.objects.get(id=context['form_entry_id'])
+
+        return context
+
+    def post(self, request, **kwargs):
+        context = self.get_context_data(**kwargs)
+        context['survey'].published = True
+        context['survey'].save()
+
+        return redirect('survey-list')
```

Comment:
When I try to publish a survey I get this error:

```
Internal Server Error: /surveys/publish/4/
Traceback (most recent call last):
  [...]
  File "/home/datamade/.virtualenvs/just-spaces/lib/python3.6/site-packages/django/urls/resolvers.py", line 622, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'survey-list' not found. 'survey-list' is not a valid view function or pattern name.
```

I think the right URL name might be `surveys-list`? In any case, this seems like a good opportunity for a test that would `POST` to publish a survey and confirm that the response returns 200.

Reply:
Oops sorry about that, you're right about `surveys-list`! I've added a test in the last commit to test survey publishing, but I can't figure out why the last assertion isn't passing. 

Could we pair on this tomorrow?

Reply:
Sure thing!

Reply:
Took a quick look at this because I had an idea of what might be happening. I think you need to refresh the `survey_form_entry` object from the DB in order to see the updated attribute. Docs: https://docs.djangoproject.com/en/dev/ref/models/instances/#django.db.models.Model.refresh_from_db

I was able to confirm that `survey_form_entry.published` was `True` after calling `survey_form_entry.refresh_from_db()`.

Reply:
Wow interesting!! Thanks, this worked.